### DISCUSSION TITLE
Suggestion for Xbox-Style controller that has working paddles/extra buttons?

### DISCUSSION THREAD

**Discussion by Mindus-Amitiel-Debsin** [sezanzeb/input-remapper#601] [General]
Hello, all!

I have been using Input Remapper for a while, and I'm really appreciative! Thanks for making it. I primarily use Pop!_OS, and I'm really happy we have such great software available.

**My Q is this:** _What controller can I buy (Xbox-Style) that has working paddles for use in Input-Remapper?
I would want to use them to bind to a Discord PTT button, and other keyboard buttons._

I've bought at least 5 different controllers in the past couple years and I've been so disappointed in them, especially the ones that have back buttons or paddles that don't work with Input-Remapper.

**Second Q:** _Are there new input drivers that I can install that will increase the function of Input-Remapper, like XpadNeo?_

Thanks for reading!

**Comment by jonasBoss** [comment]
Did you try the beta branch? That should work with all the buttons and axis.
There is this #572 Issue which limits some axis, depending on the controller, but they should all work to trigger some buttons.

**Reply by Mindus-Amitiel-Debsin** [reply]
Thanks for the reply! 
I don't mean to be a downer, but I don't have access to the beta. I'm on Pop!_OS and searching apt for input-remapper does not show a beta. I also did some google searches and could not find an AUR helper for Pop!_OS. 
What's the timeline until the next update?  =)

**Reply by jonasBoss** [reply]
No worries. I know, installing beta is currently a bit more involved as there is no ready to use beta package. But I think the manual installation should work on Pop_OS: https://github.com/sezanzeb/input-remapper/tree/beta#ubuntudebian
If you are not comfortable with the manual installation you have to wait for the 2.0 release. I think that might be ready in the coming months, but that depends how much time @sezanzeb has.

**Reply by jonasBoss** [reply]
That should work:
```bash
sudo apt install git python3-setuptools gettext
git clone https://github.com/sezanzeb/input-remapper.git
cd input-remapper && git checkout beta && ./scripts/build.sh
sudo apt install ./dist/input-remapper-1.6.0-beta.deb
```

**Reply by sezanzeb** [reply]
> I think that might be ready in the coming months, but that depends how much time @sezanzeb has.

From 11.02. on I have more time again.
- I want to finish https://github.com/sezanzeb/input-remapper/pull/578 by resolving the various TODOs in there.
- I (or anyone else who wants to do it) will then continue with https://github.com/sezanzeb/input-remapper/issues/495, which looks comparatively trivial.
- Afterwards it's only small stuff. Also see https://github.com/sezanzeb/input-remapper/projects/3

I wish to finish everything in February. I'd love to release 2.0 soon afterwards, once the changes from #578 have been tested by a few people.
